hvcc/generators/c2wwise/c2wwise.py

```diff
@@ -43,112 +42,68 @@ def compile(
         copyright: Optional[str] = None,
         verbose: Optional[bool] = False
     ) -> Dict:
-
         tick = time.time()
 
         in_parameter_list = externs["parameters"]["in"]
         out_parameter_list = externs["parameters"]["out"]
         event_list = externs["events"]["in"]
+        out_event_list = externs["events"]["out"]
         table_list = externs["tables"]
-
-        out_dir = os.path.join(out_dir, "wwise")
         patch_name = patch_name or "heavy"
-
         copyright_c = copyright_manager.get_copyright_for_c(copyright)
         copyright_xml = copyright_manager.get_copyright_for_xml(copyright)
 
-        wwise_sdk_version = "2017.2.2.6553"
+        templates_dir = os.path.join(os.path.dirname(__file__), "templates")
+        is_source_plugin = num_input_channels == 0
+        plugin_type = "Source" if is_source_plugin else "FX"
+        plugin_id = filter_plugin_id(patch_name)
 
+        out_dir = os.path.join(out_dir, "wwise")
         if not os.path.exists(out_dir):
             os.makedirs(out_dir)
 
-        templates_dir = os.path.join(os.path.dirname(__file__), "templates")
-        plugin_type = "Source" if num_input_channels == 0 else "FX"
-        plugin_id = filter_plugin_id(patch_name)
-
         env = jinja2.Environment()
-        env.filters["xcode_build"] = filter_xcode_build
-        env.filters["xcode_fileref"] = filter_xcode_fileref
         env.loader = jinja2.FileSystemLoader(
             encoding="utf-8-sig",
             searchpath=[templates_dir])
-
         try:
+            channel_config = None
+            if num_output_channels == 0:
+                # Workaround to support patches that don't have any outputs,
+                # e.g. that only ever set RTPCs or post events
+                channel_config = 'AK_SPEAKER_SETUP_MONO'
+            elif num_output_channels == 1:
+                channel_config = 'AK_SPEAKER_SETUP_MONO'
+            elif num_output_channels == 2:
+                channel_config = 'AK_SPEAKER_SETUP_STEREO'
+            elif num_output_channels == 6:
+                channel_config = 'AK_SPEAKER_SETUP_5POINT1'
+            elif num_output_channels == 8:
+                channel_config = 'AK_SPEAKER_SETUP_7POINT1'
+            elif num_output_channels == 12:
+                channel_config = 'AK_SPEAKER_SETUP_DOLBY_7_1_4'
+            else:
+                raise Exception('Wwise plugins support only mono, stereo, 5.1, 7.1, '
+                                'and 7.1.4 (Atmos) configurations, number of [dac~] channels should be appropriate')
+
             if plugin_type == "FX":
-                if num_input_channels > 2:
-                    raise Exception("Wwise FX plugins support a maximum of 2 channels (i.e. [adc~ 1] or [adc~ 1 2]).")
                 if num_input_channels != num_output_channels:
```

Comment:
Just to confirm: this is still always the case? So it's not possible to have the plugin do any up/down-mixing?
An FX always sits in the path of same number input as output channels.